### ISSUE TITLE
Training and Test-set separation

### ISSUE THREAD

**Issue by davids91** [davids91/rafko#44]
Optimizer needs to handle two kinds of data aggregates: Test set and Training set.
